File: Python/lab13/ProxyServiceSite.py
from IServiceSite import IServiceSite
from ServiceSite import ServiceSite

class ProxyServiceSite(IServiceSite):
    banned_login = ["banned", "pomer", "ne pomer"]

    def __init__(self):
        self.not_banned_login = []
        self.service_site = ServiceSite()

    def connection_by_login(self, login):
        if login in self.banned_login:
            print("You are banned\n")
        else:
            if login in self.not_banned_login:
                self.service_site.connection_by_login(login)
            else:
                self.not_banned_login.append(login)
                print("Your registry successful and authorization continues")
                self.service_site.connection_by_login(login)

File: Java/lab13/src/main/java/org/example/ProxyServiceSite.java
package org.example;

import java.util.ArrayList;
import java.util.List;

public class ProxyServiceSite implements IServiceSite{
    private static final List<String> bannedLogin;
    private final List<String> notBannedLogin=new ArrayList<>();
    private final ServiceSite serviceSite=new ServiceSite();
    static {
        bannedLogin=new ArrayList<>();
        bannedLogin.add("banned");
        bannedLogin.add("pomer");
        bannedLogin.add("ne pomer");
    }
    @Override
    public void connectionByLogin(String login) {
        if(bannedLogin.contains(login)){
            System.out.println("You are banned\n");
        }else {
            if(notBannedLogin.contains(login)){
                serviceSite.connectionByLogin(login);
            }else {
                notBannedLogin.add(login);
                System.out.println("Your regestry succesfull and authorisation continue");
                serviceSite.connectionByLogin(login);
            }
        }
    }
}
